x/dsmr/node.go

```diff
@@ -183,6 +185,9 @@ func (n *Node[T]) BuildChunk(
 		// we have duplicates
 		return ErrDuplicateChunk
 	}
+	if err := n.storage.CheckRateLimit(chunk); err != nil {
+		return fmt.Errorf("failed to meet chunk rate limits threshold : %w", err)
+	}
```

Comment:
hmm makes sense that we rate limit ourselves and fail early rather than broadcast to peers that we know will rate limit us